### ISSUE TITLE
systemsettings5 crash on launch on release-21.05

### ISSUE THREAD

**Issue by mdevlamynck** [NixOS/nixpkgs#124365]
**Describe the bug**
`systemsettings5` crashes on launch on the branch `release-21.05`.

**To Reproduce**
1. Build a system using the `release-21.05` branch (I was using `bc71efad3eac3c6d15c2fc68ac9acef919d0fa73` to be precise) and KDE 5.
2. Start `systemsettings5`.

**Expected behavior**
`systemsettings5` should start and run normally.

**Screenshots**
Console log when starting `systemsettings5`:
```
file:///nix/store/pbrq4vmygg6lbs20wvwdhrhck1y6fgac-systemsettings-5.21.5-bin/share/kpackage/genericqml/org.kde.systemsettings.sidebar/contents/ui/main.qml:23:1: Type Kirigami.ApplicationItem unavailable 
     Kirigami.ApplicationItem { 
     ^
file:///nix/store/n99y3f5124n5cc38bnvwz67qbdzzjbng-kirigami2-5.80.0/lib/qt-5.15.2/qml/org/kde/kirigami.2/ApplicationItem.qml:103:5: Type PageRow unavailable 
         PageRow { 
         ^
file:///nix/store/n99y3f5124n5cc38bnvwz67qbdzzjbng-kirigami2-5.80.0/lib/qt-5.15.2/qml/org/kde/kirigami.2/PageRow.qml:436:5: Type QQC2.StackView unavailable 
         QQC2.StackView { 
         ^
file:///nix/store/x3fk82mlnrcdkq4k78dfa2cszg61spgi-qtquickcontrols2-5.15.2-bin/lib/qt-5.15.2/qml/QtQuick/Controls.2/qmldir: plugin cannot be loaded for module "QtQuick.Controls": Cannot install type 'VerticalHeaderView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'HorizontalHeaderView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install element 'SplitHandle' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'SplitView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install element 'Overlay' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'MenuBarItem' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'MenuBar' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ActionGroup' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Action' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ScrollView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'DelayButton' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ToolSeparator' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'RoundButton' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'MenuSeparator' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'DialogButtonBox' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Dialog' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Tumbler' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ToolTip' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ToolButton' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ToolBar' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'TextField' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'TextArea' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'TabButton' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'TabBar' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'SwitchDelegate' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Switch' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'SwipeView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'SwipeDelegate' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'StackView' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'SpinBox' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Slider' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ScrollIndicator' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ScrollBar' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'RangeSlider' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'RadioDelegate' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'RadioButton' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ProgressBar' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Popup' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Pane' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'PageIndicator' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Page' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'MenuItem' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Menu' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Label' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ItemDelegate' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'GroupBox' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Frame' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Drawer' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Dial' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Control' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Container' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ComboBox' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'CheckDelegate' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'CheckBox' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ButtonGroup' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'Button' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'BusyIndicator' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'ApplicationWindow' into protected module 'QtQuick.Controls' version '2'
<Unknown File>: Cannot install type 'AbstractButton' into protected module 'QtQuick.Controls' version '2'
"file:///nix/store/pbrq4vmygg6lbs20wvwdhrhck1y6fgac-systemsettings-5.21.5-bin/share/kpackage/genericqml/org.kde.systemsettings.sidebar/contents/ui/main.qml:23:1: Type Kirigami.ApplicationItem unavailable"
"file:///nix/store/n99y3f5124n5cc38bnvwz67qbdzzjbng-kirigami2-5.80.0/lib/qt-5.15.2/qml/org/kde/kirigami.2/ApplicationItem.qml:103:5: Type PageRow unavailable"
"file:///nix/store/n99y3f5124n5cc38bnvwz67qbdzzjbng-kirigami2-5.80.0/lib/qt-5.15.2/qml/org/kde/kirigami.2/PageRow.qml:436:5: Type QQC2.StackView unavailable"
"file:///nix/store/x3fk82mlnrcdkq4k78dfa2cszg61spgi-qtquickcontrols2-5.15.2-bin/lib/qt-5.15.2/qml/QtQuick/Controls.2/qmldir: plugin cannot be loaded for module \"QtQuick.Controls\": Cannot install type 'VerticalHeaderView' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'HorizontalHeaderView' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install element 'SplitHandle' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'SplitView' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install element 'Overlay' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'MenuBarItem' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'MenuBar' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'ActionGroup' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Action' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'ScrollView' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'DelayButton' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'ToolSeparator' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'RoundButton' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'MenuSeparator' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'DialogButtonBox' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Dialog' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Tumbler' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'ToolTip' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'ToolButton' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'ToolBar' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'TextField' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'TextArea' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'TabButton' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'TabBar' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'SwitchDelegate' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Switch' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'SwipeView' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'SwipeDelegate' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'StackView' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'SpinBox' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Slider' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'ScrollIndicator' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'ScrollBar' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'RangeSlider' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'RadioDelegate' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'RadioButton' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'ProgressBar' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Popup' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Pane' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'PageIndicator' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Page' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'MenuItem' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Menu' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Label' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'ItemDelegate' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'GroupBox' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Frame' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Drawer' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Dial' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Control' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Container' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'ComboBox' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'CheckDelegate' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'CheckBox' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'ButtonGroup' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'Button' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'BusyIndicator' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'ApplicationWindow' into protected module 'QtQuick.Controls' version '2'"
"<Unknown File>: Cannot install type 'AbstractButton' into protected module 'QtQuick.Controls' version '2'"
Fatal error while loading the sidebar view qml component
[1]    2898 IOT instruction (core dumped)  systemsettings5
```

**Additional context**
I'm using flakes and the `release-21.05` branch to build my system.

**Notify maintainers**

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.21, NixOS, 21.05.20210525.bc71efa (Okapi)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210503_6d2553a`
 - channels(root): `"nixos-20.09.4169.ac60476ed94, unstable-21.05pre291486.900115a4f7f"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs
 ```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by peterhoeg** [comment]
Is `qqc2-desktop-style` in the global environment? Add it if it isn't.

Can you check if `QT_QUICK_CONTROLS_STYLE` is set?

If it is, unset it and launch `systemsettings5`.

If it isn't, try setting it to `org.kde.desktop` 

Any difference?

**Comment by mdevlamynck** [comment]
It doesn't seem to change anything unfortunately. I'm bisecting to try to find which commit introduced the bug.

**Comment by mdevlamynck** [comment]
Seems it has been fixed on the latest `release-21.05`, I don't reproduce the crash anymore.
